std/container/dlist.d

```diff
@@ -204,6 +205,15 @@ struct DList(T)
             this._payload = move(_payload);
         }
 
+        this (U)(BaseNode _base, U _payload) if (is(U == const(T)) ^ is(U == immutable(T)))
+        {
+            import std.algorithm.mutation : move;
+
+            this._base = _base;
+            temp = cast(T)_payload;
```

Comment:
Where does this `temp` come from?!

Reply:
This is NOT resolved, as it means this code isn't tested at all.

Reply:
Since `const(int)` is quite obviously not mutable, the code in the unittest calls this overload of the constructor, and it works as intended, seeing that the test passes.  Could you please explain how this code isn't tested?  Furthermore, I have already removed `temp` in the most recent commit.

Reply:
How could `temp = cast(T)_payload` ever have compiled?

Reply:
I see.  I have now removed that entire section of the code.  I should have realized  `const(T)` and `immutable(T)` implicitly convert to `T`.